keyboards/kprepublic/bm40hsrgb/keymaps/43keys/keymap.c

```diff
@@ -0,0 +1,97 @@
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_Base] = LAYOUT_planck_mit(
+        KC_Q,        KC_W,      KC_E,           KC_R,          KC_T,    KC_NO, KC_NO,       KC_Y,       KC_U,      KC_I,      KC_O,         KC_P,
+      HOME_A,      HOME_S,    HOME_D,         HOME_F,          KC_G,    KC_NO, KC_NO,       KC_H,     HOME_J,    HOME_K,    HOME_L,      HOME_QU,
+        KC_Z,        KC_X,      KC_C,           KC_V,          KC_B,    KC_NO, KC_NO,       KC_N,       KC_M,   KC_COMM,    KC_DOT,      KC_SLSH,
+       KC_NO,       KC_NO,     KC_NO,           NAVI,       Lay_SPC,        KC_NO,       KC_BSPC,       NUMB,     KC_NO,     KC_NO,        KC_NO
+      ),
+  [_Navi] = LAYOUT_planck_mit(
+      KC_WH_L,    KC_WH_D,   KC_WH_U,        KC_WH_R,        KC_ESC,    KC_NO, KC_NO,       KC_DEL,   KC_HOME,  KC_PGDN,   KC_PGUP,      KC_END,
+       HOME_A,     HOME_S,    HOME_D,         HOME_F,        KC_TAB,    KC_NO, KC_NO,       KC_ENT,    HOME_J,   HOME_K,    HOME_L,     HOME_QU,
+      KC_MS_L,    KC_MS_D,   KC_MS_U,        KC_MS_R,       KC_BTN1,    KC_NO, KC_NO,      KC_BTN2,   KC_LEFT,  KC_DOWN,     KC_UP,     KC_RGHT,    
+        KC_NO,      KC_NO,     KC_NO,          KC_NO,       Lay_SPC,        KC_NO,         KC_BSPC,     KC_NO,    KC_NO,     KC_NO,       KC_NO
+      ),  
+  [_Numb] = LAYOUT_planck_mit(
+      KC_LBRC,        KC_7,     KC_8,          KC_9,        KC_RBRC,    KC_NO, KC_NO,      KC_DEL,      KC_NO,     KC_NO,     KC_NO,       KC_NO,
+      KC_SCLN,        KC_4,     KC_5,          KC_6,         KC_EQL,    KC_NO, KC_NO,      KC_ENT,     HOME_J,    HOME_K,    HOME_L,     HOME_QU,
+       KC_GRV,        KC_1,     KC_2,          KC_3,        KC_BSLS,    KC_NO, KC_NO,       KC_NO,      KC_NO,     KC_NO,     KC_NO,       KC_NO,
+        KC_NO,       KC_NO,     KC_NO,         KC_0,        KC_MINS,        KC_NO,        KC_BSPC,      KC_NO,     KC_NO,     KC_NO,       KC_NO
+      ),  
+  [_Func] = LAYOUT_planck_mit(
+      KC_F12,     KC_F7,     KC_F8,           KC_F9,       KC_PSCR,    KC_NO, KC_NO,        RESET,      KC_NO,     KC_NO,     KC_NO,     KC_NO,
+      KC_F11,     KC_F4,     KC_F5,           KC_F6,       KC_SCRL,    KC_NO, KC_NO,      KC_CAPS,     HOME_J,    HOME_K,    HOME_L,   HOME_QU,
+      KC_F10,     KC_F1,     KC_F2,           KC_F3,       KC_PAUS,    KC_NO, KC_NO,       KC_INS,      KC_NO,     KC_NO,     KC_NO,     KC_NO,
+      KC_NO,       KC_NO,     KC_NO,          KC_NO,       Lay_SPC,         KC_NO,        KC_BSPC,      KC_NO,     KC_NO,     KC_NO,     KC_NO
+      ),
+      
+};
+
+// color by layer
+layer_state_t layer_state_set_user(layer_state_t state) {
+    uint8_t layer = biton32(state);
+    switch (layer) {
+        case _Navi:
+            rgblight_mode(11);
+            break;
+        case _Numb:
+            rgblight_mode(5);
+            break;
+        case _Func:
+            rgblight_mode(1);
+            break;
+        default:
+            rgblight_mode(22);
```

Comment:
```suggestion
            rgblight_mode_noeeprom(11);
            break;
        case _Numb:
            rgblight_mode_noeeprom(5);
            break;
        case _Func:
            rgblight_mode_noeeprom(1);
            break;
        default:
            rgblight_mode_noeeprom(22);
```
Also, I would HIGHLY recommend using the animation name, rather than just a number.

Reply:
If I knew the name of animation, I would but I have no idea of how to check it.